processing/src/main/java/org/apache/carbondata/processing/util/DeleteLoadFolders.java

```diff
@@ -133,16 +138,23 @@ public static boolean deleteLoadFoldersFromFileSystem(
       for (LoadMetadataDetails oneLoad : details) {
         if (checkIfLoadCanBeDeleted(oneLoad, isForceDelete)) {
           String path = getSegmentPath(absoluteTableIdentifier, 0, oneLoad);
-          boolean deletionStatus = physicalFactAndMeasureMetadataDeletion(path);
-          if (deletionStatus) {
-            isDeleted = true;
-            oneLoad.setVisibility("false");
-            LOGGER.info("Info: Deleted the load " + oneLoad.getLoadName());
+          ICarbonLock segmentLock = CarbonLockFactory.getCarbonLockObj(absoluteTableIdentifier,
```

Comment:
1) Please add a log for lock acquire status
2) Please check the latest status if the current in progress is just completed. If so, we will accidentally delete completed load. 
3) Can we use lock with out reties as this work is only for cleanup and retries may make clean files command very slow due to new lock.

Reply:
logic is divided